File: JavaScript/day2.js
const fs = require("fs");
var path = require('path');
var text = fs.readFileSync("./Inputs/day2.txt",
{encoding:'utf8', flag:'r'});
var data = text.split('\r\n')

directions = []
data.forEach(element => {
    directions.push(element.split(' '))
});

horizontal = 0
depth = 0

directions.forEach(element => {
    if(element[0]=='forward'){
        horizontal += Number(element[1])
    }
    else if(element[0]=='down'){
        depth += Number(element[1])
    }
    else if(element[0]=='up'){
        depth -= Number(element[1])
    }
});

console.log('horizontal: ' + horizontal)
console.log('depth: ' + depth)
console.log('first answer: ' + depth*horizontal)

horizontal = 0
depth = 0
aim = 0

directions.forEach(element => {
    if(element[0]=='forward'){
        horizontal += Number(element[1])
        depth += Number(element[1]) * aim
    }
    else if(element[0]=='down'){
        aim += Number(element[1])
    }
    else if(element[0]=='up'){
        aim -= Number(element[1])
    }
});

console.log('horizontal: ' + horizontal)
console.log('depth: ' + depth)
console.log('aim: ' + aim)
console.log('second answer: ' + depth*horizontal)

File: Python/day2.py
import pathlib
filepath = pathlib.Path(__file__).parent.parent.joinpath("Inputs/day2.txt")
raw_data = open(filepath).read()
data = raw_data.split('\n')
directions = [i.split() for i in data]

horizontal = 0
depth = 0

for i in directions:
    if(i[0] ==  'forward'):
        horizontal += int(i[1])
    elif(i[0] ==  'down'):
        depth += int(i[1])
    elif(i[0] ==  'up'):
        depth -= int(i[1])
print('Horizontal: ' + str(horizontal))
print('Depth: ' + str(depth))
print('First Answer: ' + str(horizontal*depth))

aim = 0
depth = 0
horizontal = 0

for i in directions:
    if(i[0] ==  'forward'):
        horizontal += int(i[1])
        depth += int(i[1])*aim
    elif(i[0] ==  'down'):
        aim += int(i[1])
    elif(i[0] ==  'up'):
        aim -= int(i[1])
print('Depth: ' + str(depth))
print('aim: ' + str(aim))
print('Second Answer: ' + str(horizontal*depth))